mne/decoding/tests/test_time_gen.py

```diff
@@ -70,7 +70,15 @@ def test_generalization_across_time():
     assert_equal("<GAT | fitted, start : -0.200 (s), stop : 0.499 (s), no "
                  "prediction, no score>", '%s' % gat)
     assert_equal(gat.ch_names, epochs.ch_names)
+    # test different predict function:
+    gat.decision_function(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 1))
+    gat.predict_proba(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 2))
```

Comment:
If we want to follow sklearn API, we could avoid storing y_proba, and
provide and example to score a y_proba by externalizing the scoring
function.

e.g

```
y_proba = gat.predict_proba(epochs)
score_proba = gat_score(y_proba)
# and to plot it back to normal:
gat.scores_ = score_proba
gat.plot()
```

WDYT?

On 8 January 2016 at 17:45, jona-sassenhagen <notifications@github.com>
wrote:

> In mne/decoding/tests/test_time_gen.py
> <https://github.com/mne-tools/mne-python/pull/2755#discussion_r49248023>:
>
> > @@ -70,7 +70,15 @@ def test_generalization_across_time():
> >      assert_equal("<GAT | fitted, start : -0.200 (s), stop : 0.499 (s), no "
> >                   "prediction, no score>", '%s' % gat)
> >      assert_equal(gat.ch_names, epochs.ch_names)
> > +    # test different predict function:
> > +    gat.decision_function(epochs)
> > +    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 1))
> > +    gat.predict_proba(epochs)
> > +    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 2))
>
> In the sklearn examples I can find, predicted probabilities aren't stored
> in the object, but in an external variable. I think it would here be more
> natural to store them in the gat object though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2755/files#r49248023>.
>